### PR TITLE
Treat 'prefer' version constraints as recommendations

### PR DESCRIPTION
**Pull request by bigdaz**
This PR introduces a difference in the way that 'prefer' constraints are resolved in the presence of other "strong" constraints. Now, a 'prefer' constraint is only weakly honoured, and will not cause the resolution of a version that conflicts with a 'strictly' or 'require' constraint. (Note that 'org:foo:1.0' defines a 'require' version constraint).

### CODE REVIEW

File: subprojects/core-api/src/main/java/org/gradle/api/artifacts/MutableVersionConstraint.java [hunk 1]
```diff
@@ -48,6 +48,7 @@
     /**
      * Sets the required version of this module. Any other version constraints will be overriden.
      * @param version the preferred version of this module
+     * @since 4.11
```
Comment (ljacomet) [inline subprojects/core-api/src/main/java/org/gradle/api/artifacts/MutableVersionConstraint.java]: Needs to be `4.10`

**Comment by ljacomet** [comment]
Also you have a DCO issue in the commit before last, probably isn't signed properly.

File: subprojects/core-api/src/main/java/org/gradle/api/artifacts/MutableVersionConstraint.java (hunk 1 above)
Comment (bigdaz) [reply subprojects/core-api/src/main/java/org/gradle/api/artifacts/MutableVersionConstraint.java]: Are you sure? Assuming this PR is merged into `master`, then the first version this method will appear is `4.11` (or 5.0, I guess).

Comment (ljacomet) [reply subprojects/core-api/src/main/java/org/gradle/api/artifacts/MutableVersionConstraint.java]: I am wrong indeed, had in mind this was going to `release` when we explicitly said it was not, sorry.

**Pull request merged**